pkg/util/test/util.go

```diff
@@ -643,6 +643,74 @@ func validateImageTag(tag, namespace string, listOptions map[string]string) erro
 	return nil
 }
 
+func validateMonitoring(cluster *corev1alpha1.StorageCluster, timeout, interval time.Duration) error {
+	if cluster.Spec.Monitoring != nil && *cluster.Spec.Monitoring.EnableMetrics {
```

Comment:
We are deprecating spec.monitoring.enableMetrics flag. Instead we should use `spec.monitoring.prometheus.exportMetrics`. In the interim let's change this condition to -
```
if cluster.Spec.Monitoring != nil &&
  (*cluster.Spec.Monitoring.EnableMetrics || 
    (cluster.Spec.Monitoring.Prometheus != nil && cluster.Spec.Monitoring.Prometheus.ExportMetrics)) { ... }
```